R/spread.R

```diff
@@ -94,14 +94,21 @@ spread_.data.frame <- function(data, key_col, value_col, fill = NA,
   if (!is.na(fill)) {
     ordered[is.na(ordered)] <- fill
   }
-  if (convert) {
-    ordered <- type.convert(ordered, as.is = TRUE)
-  }
 
   dim(ordered) <- c(attr(row_id, "n"), attr(col_id, "n"))
   colnames(ordered) <- as.character(col_labels[[1]])
+  ordered <- as.data.frame.matrix(ordered, stringsAsFactors = FALSE)
```

Comment:
(It's my hand-written C code that does the same thing as `as.data.frame.matrix()` but is much faster.